### PR TITLE
Add Java SDK for DappLooker

### PR DESCRIPTION
**Pull request by 0xSumitBanik**
### Overview

This pull request introduces the Java Software Development Kit (SDK) for DappLooker under the `java` directory. The SDK empowers developers to effortlessly access reliable Web3 data and popular Dapp charts through the DappLooker Analytics platform. With seamless integration into Java projects, the SDK opens up new possibilities for exploring blockchain insights. 🚀☕️

### CODE REVIEW

File: java/src/main/java/org/dapplooker/DappLookerChartsAPI.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+        // Check if the output format is supported.
+        if (outputFormat != null && !ChartConstants.supportedFormatType.contains(outputFormat)) {
+            throw new IllegalArgumentException("Invalid output format: " + outputFormat +
+                    ". Supported formats are: " + ChartConstants.supportedFormatType);
+        }
+
+        // Build the API URL.
+        String chartApiUrl = ChartConstants.getChartDetailUrl;
+        String fullApiUrl = chartApiUrl + "/" + chartUuid + "?api_key=" + apiKey + "&output_format=" + format;
```
Comment (realchoubey) [inline java/src/main/java/org/dapplooker/DappLookerChartsAPI.java:33]: user `fmt.Sprintf` for string formatting.

File: java/src/main/java/org/dapplooker/DappLookerChartsAPI.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            if (responseCode >= 200 && responseCode < 300) {
+                String response = "";
+                try (Scanner scanner = new Scanner(connection.getInputStream())) {
+                    while (scanner.hasNext()) {
+                        response += scanner.nextLine();
+                    }
+                }
+                Map<String, Object> jsonResponse = new HashMap<>();
+                jsonResponse.put("response", response);
+                return jsonResponse;
```
Comment (realchoubey) [inline java/src/main/java/org/dapplooker/DappLookerChartsAPI.java:51]: Can it be directly return without putting into `response` key?

File: java/src/main/java/org/dapplooker/ChartConstants.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ChartConstants {
+
+    public static final List<String> supportedFormatType = Arrays.asList("json");
+    public static final String getChartDetailUrl = "https://api.dapplooker.com/chart";
+
+}
```
Comment (realchoubey) [inline java/src/main/java/org/dapplooker/ChartConstants.java:11]: Add new line.

File: java/pom.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    <groupId>org.dapplooker</groupId>
+    <artifactId>dapplooker</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
+    </properties>
+
+</project>
```
Comment (realchoubey) [inline java/pom.xml:16]: Add new line.

File: java/.gitignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+.idea
+target/*
```
Comment (realchoubey) [inline java/.gitignore:2]: Add new line at the end.

**Review by realchoubey — COMMENTED** [review]
Add the details of chart on which you have tested this, with its response as file.

File: java/README.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+   ```java
+   // Replace 'your_api_key' with the API key you obtained
+   String apiKey = "your_api_key";
+
+   // Replace 'your_chart_uuid' with the UUID of the chart you want to retrieve data for
+   String chartUUID = "your_chart_uuid";
+
+   // Make an API call to get chart data
+    Map<String,Object> data = DappLookerChartsAPI.getChartData(chartUUID, apiKey);
```
Comment (realchoubey) [inline java/README.md:83]: Add example with `output_format` as `json`, and also add sample response with and without output format.

**Comment by 0xSumitBanik** [comment]
@realchoubey , The codebase has been updated and the README has also been updated with the example.

**Review by realchoubey — APPROVED** [review]

**Pull request merged**